process/block/metablock.go

```diff
@@ -1375,8 +1365,9 @@ func (mp *metaProcessor) receivedShardHeader(shardHeaderHash []byte) {
 			"metaFinalityAttestingRound", mp.epochStartTrigger.EpochFinalityAttestingRound())
 	}
 
-	if mp.isHeaderOutOfRange(shardHeader, shardHeaderPool) || isShardHeaderWithOldEpochAndBadRound {
-		shardHeaderPool.RemoveHeaderByHash(shardHeaderHash)
+	headers := mp.dataPool.Headers()
```

Comment:
Remove this line and keep the lines between 1309-1312

Reply:
done.